src/app.js

```diff
@@ -1,26 +1,65 @@
 const getJobs = require("./getJobs");
+const getToken = require("./getToken")
+const getCookie = require("./getCookie")
 const express = require("express");
 const app = express();
-const port = 8081;
+const config = require('./config');
+const {host, port, AppID, APP_SECRET} = config;
+// const port = 8081;
+// const host = `http://localhost:${port}`
 const path = require("path");
+// const AppID = "7c0c8e717c744b103b263ef8b89d17cc429e6cfab0cddb8b3df21b66e21b45c1";
+// const APP_SECRET = "1bb0b4cf625557ee3260a756dc63bbc8544247dacd0685513fd9cf87e7f66afb";
 
 app.use(express.static(path.join(__dirname)));
 
 app.set("view engine", "ejs");
 
-//app.get("redirect", (req,res) => res.render("pages/redirect"))
+app.get("/redirect/:id", (req, res) => {
+  res.redirect(
+    `https://gitlab.com/oauth/authorize?` +
+      `&client_id=${AppID}` +
+      `&redirect_uri=http://${host}:${port}/redirect` +
```

Comment:
`http` should also be passed from env, coz local is http, the remote one must be https.

You can simply put the whole `http://${host}:${port}` as env, no need to specify them one by one.